### PR TITLE
Authentication: user entity

### PR DESCRIPTION
**Pull request by davidozhang**

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -13,4 +13,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 7e6419db4fcb749f553c2e25830be7c869d3d825
 
-COCOAPODS: 1.8.4
+COCOAPODS: 1.9.1
```
Comment (davidozhang) [inline Podfile.lock:16]: lmao i need to update my cocoapods, we keep flipping between 1.84 and 1.9.1 😂

Comment (misspia) [reply Podfile.lock:16]: omg so this is why I always have to run an install

File: SnapWar/Entities/SWUserEntity.swift [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class SWUserEntity {
+    
+    static let shared = SWUserEntity()
+    
+    var externalId: String = ""
```
Comment (davidozhang) [inline SnapWar/Entities/SWUserEntity.swift]: don't initialize these to empty string, instead just declare them as type `String?` and set them via a static factory method, something like: `func fromLoginUserInfo(externalId: String, displayName: String, avatarUrl: String?) -> SWUserEntity`

File: SnapWar/ViewControllers/SWAuthViewController.swift [hunk 3]
```diff
@@ -47,21 +44,48 @@ class SWAuthViewController: UIViewController {
     }
     
     fileprivate func fetchUserInfo() {
-        let graphQLQuery = "{me{displayName, bitmoji{avatar}, externalId}}"
-        
-        SCSDKLoginClient.fetchUserData(
-                withQuery: graphQLQuery,
-                variables: nil,
-                success: { userInfo in
+        print("fetchUserInfo")
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWAuthViewController.swift]: nit: can remove this

File: SnapWar/ViewControllers/SWAuthViewController.swift [hunk 4]
```diff
@@ -47,21 +44,48 @@ class SWAuthViewController: UIViewController {
...
+           let graphQLQuery = "{me{displayName, bitmoji{avatar}, externalId}}"
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWAuthViewController.swift]: nit: fix the indentations :)

File: SnapWar/ViewControllers/SWGalleryViewController.swift [hunk 5]
```diff
@@ -6,6 +6,8 @@ class SWGalleryViewController: UIViewController {
         super.viewDidLoad()
 
         view?.backgroundColor = UIColor.white
+
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWGalleryViewController.swift]: nit: undo this change :)

File: SnapWar/ViewControllers/SWMainNavViewController.swift [hunk 6]
```diff
@@ -8,6 +8,13 @@ class SWMainNavViewController: UITabBarController {
         setup()
     }
     
+    override func viewDidAppear(_ animated: Bool) {
+        if(SWUserEntity.shared.externalId == "") {
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWMainNavViewController.swift]: a helper method would be useful, such as `func isLoggedIn()`. we can just have a private boolean variable in the entity that indicates if the user is logged in or not, and this function can just return that. this variable is only set to true in `fromLoginUserInfo()` method

File: SnapWar/Entities/SWUserEntity.swift (hunk 2 above)
Comment (davidozhang) [reply SnapWar/Entities/SWUserEntity.swift]: in general, we should use Optional for anything that might be not initialized and use try/guard to check if they're set instead of comparing with empty string. this is the best practice!

**Review by davidozhang — COMMENTED** [review]
left some comments 😄

**Comment by davidozhang** [comment]
LGTM!

**Review by misspia — APPROVED** [review]
sick teamwork 👍
